### PR TITLE
Improve Fuse read/write core logics

### PR DESCRIPTION
**Pull request by LuQQiu**
Improve Fuse read/write core logics

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInStream.java [hunk 1]
```diff
@@ -84,24 +85,29 @@ private FuseFileInStream(FileInStream inStream, long fileLength, AlluxioURI uri)
 
   @Override
   public synchronized int read(ByteBuffer buf, long size, long offset) {
+    Preconditions.checkArgument(size >= 0 && offset >= 0 && size <= buf.capacity(),
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInStream.java:88]: If a file is opened multiple times, the same iostream is shared, right?

Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInStream.java:88]: in fuse.open() we will assign an open file id
if multiple fuse.open() are issued, multiple io stream will be created
if only one open() is issued, the iostream will be used (not quite sure whether multiple threads will share the same open() or not)

**Comment by LuQQiu** [comment]
@yyongycy PTAL, thanks!

**Comment by yyongycy** [comment]
This change is good. Just have extra question regarding if the multiple fuse open will share the same

**Review by yyongycy — APPROVED** [review]
LGTM

**Pull request merged**